### PR TITLE
Remove beta tamper proofing feature flag

### PR DESCRIPTION
**Pull request by pkwarren**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -11,8 +11,6 @@ linters-settings:
       - '^exec\.Cmd$'
       - '^exec\.Command$'
       - '^exec\.CommandContext$'
-      # This should only be used by github.com/bufbuild/buf-language-server
-      - '^bufcli\.NewModuleReaderAndCreateCacheDirsWithExternalPaths'
```
Comment (pkwarren) [inline .golangci.yml:15]: The LSP will need to be updated to support new cache paths.

File: private/buf/cmd/buf/command/push/push_test.go [hunk 2]
```diff
@@ -81,34 +80,34 @@ func TestPush(t *testing.T) {
 	)
 }
 
-func TestPushCreate(t *testing.T) {
+func TestPushManifestCreate(t *testing.T) {
```
Comment (pkwarren) [inline private/buf/cmd/buf/command/push/push_test.go:83]: Retrofitted the new push --create tests to use the M+B endpoint.

File: .golangci.yml (hunk 1 above)
Comment (bufdev) [reply .golangci.yml:15]: The LSP was just a prototype and isn't being actively developed, we don't need to update it

**Review by bufdev — APPROVED** [review]

File: private/buf/cmd/buf/command/push/push_test.go [hunk 3]
```diff
@@ -221,30 +204,10 @@ func newMockPushService(t *testing.T) *mockPushService {
...
-		}
-	}
-	return connect.NewResponse(resp), nil
-}
-
-func (m *mockPushService) PushRequest() *registryv1alpha1.PushRequest {
-	m.RLock()
-	defer m.RUnlock()
-	return m.pushRequest
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("unimplemented"))
```
Comment (unmultimedio) [inline private/buf/cmd/buf/command/push/push_test.go:210]:
```suggestion
	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("Push RPC should not be called, use PushManifestAndBlobs RPC instead"))
```

File: private/bufpkg/bufapimodule/module_reader_test.go [hunk 4]
```diff
@@ -249,12 +216,7 @@ func (m *mockDownloadService) Download(
 	context.Context,
 	*connect.Request[registryv1alpha1.DownloadRequest],
 ) (*connect.Response[registryv1alpha1.DownloadResponse], error) {
-	if m.err != nil {
-		return nil, m.err
-	}
-	return connect.NewResponse(&registryv1alpha1.DownloadResponse{
-		Module: m.module,
-	}), nil
+	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("unimplemented"))
```
Comment (unmultimedio) [inline private/bufpkg/bufapimodule/module_reader_test.go:219]: Same
```suggestion
	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("Download RPC should not be called, use DownloadManifestAndBlobs RPC instead"))
```

File: private/bufpkg/bufmodule/bufmodulecache/bufmodulecache.go [hunk 5]
```diff
@@ -35,41 +31,8 @@ func NewRepositoryServiceClientFactory(clientConfig *connectclient.Config) Repos
...
-	)
-}
-
-// ModuleReaderWithExternalPaths is used to preserve the external paths
-// to the files resolved from the module cache.
-func ModuleReaderWithExternalPaths() ModuleReaderOption {
-	return func(moduleReaderOptions *moduleReaderOptions) {
-		moduleReaderOptions.allowCacheExternalPaths = true
-	}
-}
```
Comment (unmultimedio) [inline private/bufpkg/bufmodule/bufmodulecache/bufmodulecache.go:68]: Just curious, did the v1 cache allowed caching paths outside of the cacheV1 dir? 🤯

**Review by unmultimedio — APPROVED** [review]
Looking nice and clean 👍🏼

**Review by unmultimedio — APPROVED** [review]

File: private/bufpkg/bufmodule/bufmodulecache/bufmodulecache.go (hunk 5 above)
Comment (pkwarren) [reply private/bufpkg/bufmodule/bufmodulecache/bufmodulecache.go:68]: I think this was to allow the LSP implementation to get access to the underlying cache files.

**Review by unmultimedio — APPROVED** [review]

**Review by unmultimedio — APPROVED** [review]

**Review by bufdev — APPROVED** [review]

**Review by saquibmian — APPROVED** [review]
🎉

**Pull request merged**